### PR TITLE
[Utility] Adding Fisherman diagrams (Issue #12)

### LINKED ISSUE #12: [Utility] Fisherman Protocol - Charts & Diagrams
# Objective
Create sequence diagrams and flow charts for the fisherman protocol.

# Origin Document

The fisherman protocol is the core of the V1 Pocket Network. As described in the [V1 Pocket Network Utility Fisherman Protocol Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/utility#33-fisherman-protocol), there are a lot of moving parts here.

# Goals
- [ ] Make the spec clearer so that it is easier for others (new tea members or external contributors) to understand
- [ ] Preparatory work / clarification that can be used to define the interface of the "fisherman protocol module" in a follow up task
- [ ] Preparatory work / clarification that can be used to implement an isolated "fisherman protocol module" that can be developed and tested in isolation using mocks of dependencies

# Deliverables
- [ ] A sequence diagram showing the types of requests and flow of data throughout a fisherman's lifecycle using [mermaid flow charts](https://mermaid-js.github.io/mermaid/#/sequenceDiagram). Note that more than one may be necessary. For an example, see what we have in the [persistence module](https://github.com/pokt-network/pocket-network-protocol/tree/main/persistence#562-state-transition-sequence-diagram) architecture doc.
- [ ] A flow chart showing which components and modules are involved in the fisherman protocol using [mermaid flow charts](https://mermaid-js.github.io/mermaid/#/flowchart)

- [ ] Follow up tasks of things that need to be design and implemented (e.g. fast sync design, slow sync implementation, etc...)

# Non-goals
- Defining or updating the codified interface to be used by the fisherman protocol
- Implementing any part of the fisherman protocol (i.e. no code)
- Modifying the fisherman protocol

---

Creator: @Olshansk 
Co-Owners: @andrewnguyen22

### PR DESCRIPTION
**Pull request by andrewnguyen22**
# Objective
Create sequence diagrams and flow charts for the fisherman protocol.

# Origin Document

The fisherman protocol is the core of the V1 Pocket Network. As described in the [V1 Pocket Network Utility Fisherman Protocol Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/utility#33-fisherman-protocol), there are a lot of moving parts here.

# Issue

closes #12 

# Deliverables

- [x] A sequence diagram showing the types of requests and flow of data throughout a fisherman's lifecycle using [mermaid flow charts](https://mermaid-js.github.io/mermaid/#/sequenceDiagram). Note that more than one may be necessary. For an example, see what we have in the [persistence module](https://github.com/pokt-network/pocket-network-protocol/tree/main/persistence#562-state-transition-sequence-diagram) architecture doc.
-  [x] A flow chart showing which components and modules are involved in the fisherman protocol using [mermaid flow charts](https://mermaid-js.github.io/mermaid/#/flowchart)
-  [x] Follow up tasks of things that need to be design and implemented (e.g. fast sync design, slow sync implementation, etc...)

Follow Up Tasks:
- Add glossary for *Variable* terminology e.g: FishermenSalaryPool
- Provide clarity on the AAT transfer protocol
- Further demonstrate non-null sample logic and interaction
- Follow up on the *Portal* interaction with the lifecycle

### CODE REVIEW

**Comment by Olshansk** [comment]
@andrewnguyen22 Could you update the PR description to something more semantic like `[Utility] Adding Fisherman diagrams (Issue #12)`?

**Comment by andrewnguyen22** [comment]
@Olshansk bump as this affects https://github.com/pokt-network/pocket/issues/265 and https://github.com/pokt-network/pocket/issues/264

**Review by Olshansk — CHANGES_REQUESTED** [review]
Made a couple minor edits to the content + just making the diagrams easier to maintain.

I anticipate that we'll iterate, add an improve on these over time but I honestly think it already makes it 100x easier to read.

File: utility/README.md [hunk 1]
```diff
@@ -159,6 +189,18 @@ type FishermanStakeMsg interface {
 
 
 Once successfully staked in the network, Fishermen are eligible to receive monitoring requests from Applications. Within a monitoring request, a limited ApplicationAuthenticationToken and Ephemeral Private key is provided to Fishermen which enables them to make sampling requests to all of the ServiceNodes in the Session on the Application’s behalf. In addition to the AAT, the Application may optionally provide the Fishermen a list of acceptable monitoring requests to test the ServiceNodes with during the sampling. Once the Application handshake is complete, the Fisherman is effectively a monitoring client of the Application and will execute periodic sampling inline with the Sampling Protocol. 
+```mermaid
+
+graph LR
+graph LR
```
Comment (Olshansk) [inline utility/README.md:195]:
```suggestion
```

**Review by Olshansk — APPROVED** [review]
There's a small syntax error (introduced by me) where `graph LR` shows up twice, but LGTM otherwise.

**Pull request merged**